src/approaches/interactive_learning_approach.py

```diff
@@ -91,17 +109,16 @@ def learn_from_interaction_results(
         result = results[0]
         for state, response in zip(result.states, result.responses):
             assert isinstance(response, GroundAtomsHoldResponse)
+            state_annotation: Dict[str, Set[GroundAtom]] = {
```

Comment:
random question, what if this `state` already appeared somewhere else in the dataset? then wouldn't you need to update its annotation, otherwise you could give bad data to the learning?

Reply:
e.g. if the teacher left out an atom due to `ratio < 1.0` but then during online you ask about that state and the teacher says it's positive

Reply:
hmm that's true... that is annoying... I'll have to think more about this.

Reply:
well actually it might be fine. the only case where we would want to update an annotation is if we previously marked something as unlabelled, but now we know that it's actually positive or negative. right?

so i'm going to file this concern under the comment in the code:
```
                    # Here we make the (wrong in general!) assumption that
                    # unlabeled ground atoms are negative. In the future, we
                    # may want to modify this, e.g., downweight or remove
                    # the unlabeled examples once we collect enough negatives.
```

Reply:
Note that active learning can never create an "unlabeled" label. So active learning should always overwrite any unlabeled label that was created during offline learning. But unfortunately this means you have to loop through the dataset to find matching states to the one you queried?

Reply:
lol yeah we said the same thing

Reply:
the only problem is when we now know that it's actually positive, that will lead to contradictory data